dyne/ukf.py

```diff
@@ -71,7 +71,8 @@ def run_ukf(X0, P0, f, Q, measurements, n_epochs, alpha=1.0):
         for epochs, Z, h, R in measurements:
             index = np.searchsorted(epochs, k)
             if index < len(epochs) and epochs[index] == k:
-                Sigma = alpha * np.sqrt(n_states) * linalg.cholesky(P[k])
+                _, S, Vt = linalg.svd(P[k])
+                Sigma = alpha * np.sqrt(n_states) * np.diag(S**0.5) @ Vt
```

Comment:
Wait a second...

Is it even correct? We want to find a square root of `P`, but SVD deals with with products like `P P^T`. Sure, there is  a connection between Eigendecomposition and SVD, but I have a suspicion that it's done incorrectly here.

Reply:
I wrongly thought about something else. Indeed for positive-definite matrices`svd` and `eigh` give the same result.

I don't know details on the numerical methods behind that. The discussion here hints that `eigh` might be slightly preferable https://stackoverflow.com/questions/50358310/how-does-numpy-linalg-eigh-vs-numpy-linalg-svd

Conceptually I think `eigh` is better, because:

1. It takes into account the symmetry property
2. It returns symmetric structure, i. e. `U S U^-1` (the `U` is orthogonal and `U^-1 = U^T`)

Whereas SVD as far as I understand works as if the input matrix is arbitrary.

Reply:
agreed to change `P` decomposition from `svd` to `eigh` with manual zeroing small negative eigenvalues